.github/guides/STYLE.md

```diff
@@ -399,6 +400,192 @@ turn_on(power_usage = 30) // Fine!
+```
+
+This relies on "max_health", but it is not obviously clear what the source is. This will also become worse if we *do* want to change where we get the source from. This would be preferential as:
+
+```dm
+#define GET_HEALTH(source, health_percent) (##health_percent * ##source.max_health)
+```
+
+When a macro can't be hygienic, such as in the case where a macro is preferred to do something like define a variable, it should still do its best to rely only on input given to it:
+
+```dm
+#define DECLARE_MOTH(name) var/mob/living/moth/moth = new(##name)
+```
+
+...would ideally be written as...
+
+```dm
+#define DECLARE_MOTH(var_name, name) var/mob/living/moth/##var_name = new(##name)
+```
+
+As usual, exceptions exist--for instance, accessing a global like a subsystem within a macro is generally acceptable.
+
+### ...but if you must be unhygienic, try to restrict the scope.
+
+This guideline can make some code look extremely noisy if you are writing a large proc, or using the macro a large amount of times.
+
+In this case, if your macro is only used by one proc, define the macro in that proc, ideally after whatever variables it uses.
+
+```dm
+/proc/some_complex_proc()
+	var/counter = 0
+
+	#define MY_NECESSARY_MACRO counter += 5; do_something(counter);
+
+	// My complex code that uses MY_NECESSARY_MACRO here...
+
+	#undef MY_NECESSARY_MACRO
+```
+
+### Don't perform side effects/perform unhygienic actions in an unsuspecting macro
+
+Suppose we have the following macro:
+
+```dm
+#define PARTY_LIGHT_COLOR (pick(COLOR_BLUE, COLOR_RED, COLOR_GREEN))
+```
+
+When this is used, it'll look like:
+
+```dm
+set_color(PARTY_LIGHT_COLOR)
+```
+
+Because of how common using defines as constants is, this would seemingly imply the same thing! It does not look like any code should be executing at all. This code would preferably look like:
+
+```dm
+set_color(PARTY_LIGHT_COLOR())
+```
+
+...which *does* imply some work is happening.
+
+BYOND does not support `#define PARTY_LIGHT_COLOR()`, so instead we would write the define as:
+
+```dm
+#define PARTY_LIGHT_COLOR(...) (pick(COLOR_BLUE, COLOR_RED, COLOR_GREEN))
+```
+
+### `#undef` any macros you create, unless they are needed elsewhere
+
+We do not want macros to leak outside their file, this will create odd dependencies that are based on the filenames. Thus, you should `#undef` any macro you make.
+
+```dm
+// Start of corn.dm
+#define CORN_KERNELS 5
+
+// All my brilliant corn code
+
+#undef CORN_KERNELS
+```
+
+It is often preferable for your `#define` and `#undef` to surround the code that actually uses them, for instance:
+
+```dm
+/obj/item/corn
+	name = "yummy corn"
+
+#define CORN_HEALTH_GAIN 5
+
+/obj/item/corn/proc/eat(mob/living/consumer)
+	consumer.health += CORN_HEALTH_GAIN // yum
+
+#undef CORN_HEALTH_GAIN
+
+// My other corn code
+```
+
+If you want other files to use macros, put them in somewhere such as a file in `__DEFINES`.
```

Comment:
It'd be nice to mention how we intentionally hoist files like this, or why it works more particularly